projectq/backends/_awsbraket/_awsbraket.py

```diff
@@ -0,0 +1,452 @@
+            if isinstance(g, (XGate)) and get_control_count(cmd) == 1:
+                return True
+            if get_control_count(cmd) == 0:
+                if g in (Sdag, Tdag):
+                    return True
+                if isinstance(g, (Rx, Ry, Rz,
+                              XGate, YGate, ZGate,
+                              HGate, SGate, TGate,
+                              SqrtXGate, SwapGate)):
+                    return True
+            return False
+        if self.device == 'SV1':
+            if isinstance(g, (R, ZGate, YGate, XGate,
+                          SwapGate)) and get_control_count(cmd) == 1:
+                return True
+            if isinstance(g, (XGate)) and get_control_count(cmd) == 2:
+                return True
+            if get_control_count(cmd) == 0:
+                if isinstance(g, (R, Rx, Ry, Rz,
+                              XGate, YGate, ZGate,
+                              HGate, SGate, TGate,
+                              SqrtXGate, SwapGate)):
+                    # TODO: add MatrixGate to cover the unitary operation
+                    return True
+                if g in (Sdag, Tdag):
+                    return True
+            return False
+        return False
+
+    def _reset(self):
+        """ Reset all temporary variables (after flush gate). """
+        self._clear = True
+        self._measured_ids = []
+
+    def _store(self, cmd):
+        """
+        Temporarily store the command cmd.
+
+        Translates the command and stores it in a local variable (self._cmds).
+
+        Args:
+            cmd: Command to store
+        """
+        if self._clear:
+            self._probabilities = dict()
+            self._clear = False
+            self._circuit = ""
+            self._allocated_qubits = set()
+
+        # Previous to store the gate, checks availability against the device
+        if not self.is_available(cmd):
+            raise Exception('Invalid command: ' + str(cmd) +
+                            '. Please check the available commands '
+                            'for the device ' +
+                            self.device + '.')
+
+        gate = cmd.gate
+        num_controls = get_control_count(cmd)
+        gate_type = type(gate)
+
+        if gate == Allocate:
+            self._allocated_qubits.add(cmd.qubits[0][0].id)
+            return
+        elif gate == Deallocate:
+            return
+        elif gate == Barrier:
+            return
+        elif gate == Measure:
+            assert len(cmd.qubits) == 1 and len(cmd.qubits[0]) == 1
+            qb_id = cmd.qubits[0][0].id
+            logical_id = None
+            for tag in cmd.tags:
+                if isinstance(tag, LogicalQubitIDTag):
+                    logical_id = tag.logical_qubit_id
+                    break
+            # assert logical_id is not None
+            if logical_id is None:
+                logical_id = qb_id
+                self._mapper.append(qb_id)
```

Comment:
```suggestion
```

Reply:
But need to conserve the if in order to assign logical_id = qb_id and populate self._measured_ids.

There would be
```
# assert logical_id is not None
if logical_id is None:
     logical_id = qb_id
self._measured_ids += [logical_id]
```

Reply:
Right, forgot about that. You could even do that in a one-liner if you so wished to:
```suggestion
            self._measured_ids += [logical_id if logical_id is not None else qb_id]
```